### ISSUE TITLE
aws_cognito_identity_pool pool name regex expresssion is wrong

### ISSUE THREAD

**Issue by liberwang1013** [hashicorp/terraform-provider-aws#11558]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
terraform v0.12.19

### Affected Resource(s)

* aws_cognito_identity_pool

### Expected Behavior

cognito identity pool name can contain "-"

### Actual Behavior

```
Error: "identity_pool_name" must contain only alphanumeric characters and spaces
```

### Steps to Reproduce

1. Create cognito identity pool [Sample of Code](https://www.terraform.io/docs/providers/aws/r/cognito_identity_pool.html)
2. `terraform plan`

### References

https://github.com/terraform-providers/terraform-provider-aws/blob/5348ed3f1c47900f18c7f457a920de5f33cf7142/aws/validators.go#L1455
* https://docs.aws.amazon.com/cognitoidentity/latest/APIReference/API_IdentityPoolShortDescription.html

**Comment by john-delivuk-rl** [comment]
Please upvote #15773 for merge :)

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 3.15.0 of the Terraform AWS Provider, later this week. Thanks to @john-delivuk-rl for the implementation. 👍
